api/src/main/java/org/apache/iceberg/DataFile.java

```diff
@@ -96,4 +101,9 @@ default FileContent content() {
   default List<Integer> equalityFieldIds() {
     return null;
   }
+
+  @Override
+  default Map<Integer, Long> nanValueCounts() {
+    return null;
+  }
```

Comment:
I think all of the implementations will provide this method, so there should be no need for the default, right?

Reply:
Originally since V1 doesn't have this so it was used there, but as we are going to support it, I'll remove it.